File: Third_Year/DS/Practicas/P1/Método Factoria/PistaFutbol.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package DesignPatterns_FactoryMethod;

/**
 *
 * @author raulrguez
 */
public class PistaFutbol extends Pista {
    private String type;

    public PistaFutbol (float tamanio,String t){
        super(tamanio);
        this.type = t;
        this.tipo = "Fútbol";
    }

}

File: Third_Year/DS/Practicas/P1/Prototipo/PistaFutbol.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require './Pista'
module Prototipo

class PistaFutbol < Pista
  public
  def initialize(id)
    super(id)
    @players = Array.new
    @tipo = "Futbol"
  end

  def clonar
    return PistaFutbol.new(@id)
  end

  def addJugadores(jug)
    @players << jug
  end

  attr_accessor :players
end

end